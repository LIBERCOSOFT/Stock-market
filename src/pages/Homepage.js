/* eslint-disable max-len */
import React, { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { Link } from 'react-router-dom';
import { fetchStocks } from '../redux/homepage/homepage';

import allStockImg from '../components/images/all_stock.svg';
import consumerCyclical from '../components/images/consumer-svgrepo-com.svg';
import energy from '../components/images/energy-svgrepo-com.svg';
import technology from '../components/images/Technology.svg';
import industrial from '../components/images/industry-svgrepo-com.svg';
import financialServices from '../components/images/financial-services.svg';
import basicMaterials from '../components/images/basic-home-svgrepo-com.svg';
import communicationServices from '../components/images/communication-svgrepo-com.svg';
import consumerDefence from '../components/images/consumer-defence.svg';
import healthCare from '../components/images/healthcare-svgrepo-com.svg';
import realEstate from '../components/images/real-estate-svgrepo-com.svg';
import utilities from '../components/images/Utilities.svg';
import industrialGoods from '../components/images/Industrial_Goods.svg';
import financial from '../components/images/Financial.svg';
import services from '../components/images/Services.svg';
import conglomerates from '../components/images/Conglomerates.svg';

const Homepage = () => {
  const allStocks = useSelector((state) => state.allStocks.data);

  const dispatch = useDispatch();

  useEffect(() => {
    if (allStocks.allAssets.length < 1) {
      dispatch(fetchStocks());
    }
  }, [dispatch, allStocks]);

  const filterOptions = [
    { query: 'Consumer Cyclical', img: consumerCyclical },
    { query: 'Energy', img: energy },
    { query: 'Technology', img: technology },
    { query: 'Industrials', img: industrial },
    { query: 'Financial Services', img: financialServices },
    { query: 'Basic Materials', img: basicMaterials },
    { query: 'Communication Services', img: communicationServices },
    { query: 'Consumer Defensive', img: consumerDefence },
    { query: 'Healthcare', img: healthCare },
    { query: 'Real Estate', img: realEstate },
    { query: 'Utilities', img: utilities },
    { query: 'Industrial Goods', img: industrialGoods },
    { query: 'Financial', img: financial },
    { query: 'Services', img: services },
    { query: 'Conglomerates', img: conglomerates },
  ];

  return (
    <div className="homepage">
      <div className="all__stock__container">
        <img src={allStockImg} alt="" />
        <div>
          <h2>All Stock Data On The Stock Market</h2>
          <span>
            {allStocks.allAssets.length.toLocaleString()}
            {' '}
            Total Stocks
          </span>
          <Link to="/listall">
            See All Stocks
            {' '}
            <i className="fa-sharp fa-solid fa-arrow-right" />
            {' '}
          </Link>
        </div>
      </div>
      <h4>Filter By Sector:</h4>
      <div className="filter__container">
        {filterOptions.map((val) => (
          <Link to={`/list/${val.query}`} key={val.query}>
            <i className="fa-sharp fa-solid fa-arrow-right" />
            <img src={val.img} alt="" />
            <h3>{val.query}</h3>
            <span>
              {allStocks[val.query].length.toLocaleString()}
              {' '}
              Stocks
            </span>
          </Link>
        ))}
      </div>
    </div>
  );
};

export default Homepage;
