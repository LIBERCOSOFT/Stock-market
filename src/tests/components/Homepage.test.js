import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../../redux/configureStore';
import Homepage from '../../pages/Homepage';

test('renders homepage component snapshot', () => {
  const { asFragment } = render(
    <Provider store={store}>
      <BrowserRouter>
        <Homepage />
      </BrowserRouter>
    </Provider>,
  );
  expect(asFragment()).toMatchSnapshot();
});
