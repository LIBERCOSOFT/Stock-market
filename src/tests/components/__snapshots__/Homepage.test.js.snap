// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders homepage component snapshot 1`] = `
<DocumentFragment>
  <div
    class="homepage"
  >
    <div
      class="loader__container"
    >
      <span
        style="display: inherit; position: relative;"
      >
        <span
          style="position: absolute; font-size: 5px; width: 15px; height: 15px; background: rgb(7, 11, 18); border-radius: 50%; animation: react-spinners-PropagateLoader-propogate-0 1.5s infinite; animation-fill-mode: forwards;"
        />
        <span
          style="position: absolute; font-size: 5px; width: 15px; height: 15px; background: rgb(7, 11, 18); border-radius: 50%; animation: react-spinners-PropagateLoader-propogate-1 1.5s infinite; animation-fill-mode: forwards;"
        />
        <span
          style="position: absolute; font-size: 5px; width: 15px; height: 15px; background: rgb(7, 11, 18); border-radius: 50%; animation: react-spinners-PropagateLoader-propogate-2 1.5s infinite; animation-fill-mode: forwards;"
        />
        <span
          style="position: absolute; font-size: 5px; width: 15px; height: 15px; background: rgb(7, 11, 18); border-radius: 50%; animation: react-spinners-PropagateLoader-propogate-3 1.5s infinite; animation-fill-mode: forwards;"
        />
        <span
          style="position: absolute; font-size: 5px; width: 15px; height: 15px; background: rgb(7, 11, 18); border-radius: 50%; animation: react-spinners-PropagateLoader-propogate-4 1.5s infinite; animation-fill-mode: forwards;"
        />
        <span
          style="position: absolute; font-size: 5px; width: 15px; height: 15px; background: rgb(7, 11, 18); border-radius: 50%; animation: react-spinners-PropagateLoader-propogate-5 1.5s infinite; animation-fill-mode: forwards;"
        />
      </span>
    </div>
    <div
      class="all__stock__container"
    >
      <img
        alt=""
        src="all_stock.svg"
      />
      <div>
        <h2>
          All Stock Data On The Stock Market
        </h2>
        <span>
          0 Total Stocks
        </span>
        <a
          href="/listall"
        >
          See All Stocks 
          <i
            class="fa-sharp fa-solid fa-arrow-right"
          />
           
        </a>
      </div>
    </div>
    <h4>
      Filter By Sector:
    </h4>
    <div
      class="filter__container"
    >
      <a
        href="/list/Consumer Cyclical"
      >
        <i
          class="fa-sharp fa-solid fa-arrow-right"
        />
        <img
          alt=""
          src="consumer-svgrepo-com.svg"
        />
        <h3>
          Consumer Cyclical
        </h3>
        <span>
          0 Stocks
        </span>
      </a>
      <a
        href="/list/Energy"
      >
        <i
          class="fa-sharp fa-solid fa-arrow-right"
        />
        <img
          alt=""
          src="energy-svgrepo-com.svg"
        />
        <h3>
          Energy
        </h3>
        <span>
          0 Stocks
        </span>
      </a>
      <a
        href="/list/Technology"
      >
        <i
          class="fa-sharp fa-solid fa-arrow-right"
        />
        <img
          alt=""
          src="Technology.svg"
        />
        <h3>
          Technology
        </h3>
        <span>
          0 Stocks
        </span>
      </a>
      <a
        href="/list/Industrials"
      >
        <i
          class="fa-sharp fa-solid fa-arrow-right"
        />
        <img
          alt=""
          src="industry-svgrepo-com.svg"
        />
        <h3>
          Industrials
        </h3>
        <span>
          0 Stocks
        </span>
      </a>
      <a
        href="/list/Financial Services"
      >
        <i
          class="fa-sharp fa-solid fa-arrow-right"
        />
        <img
          alt=""
          src="financial-services.svg"
        />
        <h3>
          Financial Services
        </h3>
        <span>
          0 Stocks
        </span>
      </a>
      <a
        href="/list/Basic Materials"
      >
        <i
          class="fa-sharp fa-solid fa-arrow-right"
        />
        <img
          alt=""
          src="basic-home-svgrepo-com.svg"
        />
        <h3>
          Basic Materials
        </h3>
        <span>
          0 Stocks
        </span>
      </a>
      <a
        href="/list/Communication Services"
      >
        <i
          class="fa-sharp fa-solid fa-arrow-right"
        />
        <img
          alt=""
          src="communication-svgrepo-com.svg"
        />
        <h3>
          Communication Services
        </h3>
        <span>
          0 Stocks
        </span>
      </a>
      <a
        href="/list/Consumer Defensive"
      >
        <i
          class="fa-sharp fa-solid fa-arrow-right"
        />
        <img
          alt=""
          src="consumer-defence.svg"
        />
        <h3>
          Consumer Defensive
        </h3>
        <span>
          0 Stocks
        </span>
      </a>
      <a
        href="/list/Healthcare"
      >
        <i
          class="fa-sharp fa-solid fa-arrow-right"
        />
        <img
          alt=""
          src="healthcare-svgrepo-com.svg"
        />
        <h3>
          Healthcare
        </h3>
        <span>
          0 Stocks
        </span>
      </a>
      <a
        href="/list/Real Estate"
      >
        <i
          class="fa-sharp fa-solid fa-arrow-right"
        />
        <img
          alt=""
          src="real-estate-svgrepo-com.svg"
        />
        <h3>
          Real Estate
        </h3>
        <span>
          0 Stocks
        </span>
      </a>
      <a
        href="/list/Utilities"
      >
        <i
          class="fa-sharp fa-solid fa-arrow-right"
        />
        <img
          alt=""
          src="Utilities.svg"
        />
        <h3>
          Utilities
        </h3>
        <span>
          0 Stocks
        </span>
      </a>
      <a
        href="/list/Industrial Goods"
      >
        <i
          class="fa-sharp fa-solid fa-arrow-right"
        />
        <img
          alt=""
          src="Industrial_Goods.svg"
        />
        <h3>
          Industrial Goods
        </h3>
        <span>
          0 Stocks
        </span>
      </a>
      <a
        href="/list/Financial"
      >
        <i
          class="fa-sharp fa-solid fa-arrow-right"
        />
        <img
          alt=""
          src="Financial.svg"
        />
        <h3>
          Financial
        </h3>
        <span>
          0 Stocks
        </span>
      </a>
      <a
        href="/list/Services"
      >
        <i
          class="fa-sharp fa-solid fa-arrow-right"
        />
        <img
          alt=""
          src="Services.svg"
        />
        <h3>
          Services
        </h3>
        <span>
          0 Stocks
        </span>
      </a>
      <a
        href="/list/Conglomerates"
      >
        <i
          class="fa-sharp fa-solid fa-arrow-right"
        />
        <img
          alt=""
          src="Conglomerates.svg"
        />
        <h3>
          Conglomerates
        </h3>
        <span>
          0 Stocks
        </span>
      </a>
    </div>
  </div>
</DocumentFragment>
`;
